plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3306,24 +3308,23 @@ public void testOptimize()
         assertUpdate("DROP TABLE " + tableName);
     }
 
-    private List<String> getActiveFiles(String tableName)
+    private Set<String> getActiveFiles(String tableName)
     {
         return computeActual(format("SELECT file_path FROM \"%s$files\"", tableName)).getOnlyColumn()
                 .map(String.class::cast)
-                .collect(toImmutableList());
+                .collect(toImmutableSet());
     }
 
-    private List<String> getAllDataFilesFromTableDirectory(String tableName)
+    private Set<String> getAllDataFilesFromTableDirectory(String tableName)
             throws IOException
     {
         String schema = getSession().getSchema().orElseThrow();
         Path tableDataDir = getDistributedQueryRunner().getCoordinator().getBaseDataDir().resolve("iceberg_data").resolve(schema).resolve(tableName).resolve("data");
-        try (Stream<Path> list = Files.list(tableDataDir)) {
-            return list
-                    .filter(path -> !path.getFileName().toString().matches("\\..*\\.crc"))
-                    .map(Path::toString)
-                    .collect(toImmutableList());
-        }
+        return Files.walk(tableDataDir)
```

Comment:
from `Files.walk` docs

> This method must be used within a try-with-resources statement or similar control structure to ensure that the stream's open directories are closed promptly after the stream's operations have completed.

Anyway, the changes here look as not related to the main part of the commit, while they actually are.
I'd extract this to a prep commit like "Let getAllDataFilesFromTableDirectory recurse directories"

Reply:
ok